### PR TITLE
Add codespace configuration for h5pyd

### PR DESCRIPTION
**Pull request by jreadey**
This PR enables h5pyd to be run as a github codespace.
Test suite should be runable within the codespace as well as scripts/notebooks under examples.

### CODE REVIEW

**Comment by mattjala** [comment]
Would it make sense to set up a running HSDS instance in the codespace? The convenience of the codespace might not be super helpful to a user if they still need to install and set up HSDS.

File: .hsds/data/hsdstest/home/.domain.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"owner": "admin", "acls": {"admin": {"create": true, "read": true, "update": true, "delete": true, "readACL": true, "updateACL": true}, "default": {"create": false, "read": true, "update": false, "delete": false, "readACL": false, "updateACL": false}}, "created": 1708897646.0599918, "lastModified": 1708897646.0599918}
```
Comment (mattjala) [inline .hsds/data/hsdstest/home/.domain.json:1]: It looks like the `hsds/data` folder was included by accident.

**Comment by jreadey** [comment]
hsds is installed as part of the codespace.   You can run hsinfo in the codespace terminal and see it connecting to http://localhost:5101.

**Review by mattjala — APPROVED** [review]
LGTM

File: .hsds/data/hsdstest/home/.domain.json (hunk 1 above)
Comment (jreadey) [reply .hsds/data/hsdstest/home/.domain.json:1]: removed

**Pull request merged**